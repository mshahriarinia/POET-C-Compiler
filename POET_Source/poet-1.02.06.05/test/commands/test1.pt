<* code templates *>
<code MyCodeTemplate pars=(input) />

<xform foo pars=(bar)> bar </xform>

<eval DEBUG{
       "3" : STRING;
       3 : STRING;
       "3" : ID;
        "A3" : ID;
       MyCodeTemplate#"123" : STRING;
      MyCodeTemplate#123 : MyCodeTemplate;
       3 : MyCodeTemplate;
       MyCodeTemplate#123 : MyCodeTemplate#INT;
       ("abc" "." "ext") : STRING;
       ("3" "4" "5") : (INT ...);
       3 : (INT ...);
       (3 4 5) : (INT ...);
       (3 4 5 "abc") : (INT ...);
       (3 4 5 "abc") : (_ ...);
        3 : (0 .. 2);
        3 : (0 .. 5);
        "a" : (0 .. 5);
       "a" : CODE;
       MyCodeTemplate : CODE;
       MyCodeTemplate#123 : CODE;
       ("abc" "." "ext") : CODE;
       MyCodeTemplate#123 : XFORM;
       ("abc" "." "ext") : XFORM;
       foo : XFORM; 
       "abc" : TUPLE;
       ("abc",2) : TUPLE;
        MyCodeTemplate#123 : TUPLE; 
}/>

<eval DEBUG {
       "3"=>STRING;
       a = ("3" "5");
       a  => STRING;
       3 => STRING;
       ("abc" "." "ext") => STRING;
       ("3" => INT) => STRING;
}/>

<eval DEBUG {
    a1 = (3 4 5);
    a2 = (1 2 a1); 
    a3 =  (1 2) :: a1;
    a4 = 1 :: 2 :: a1;
    car(a2);
    cdr(a2);
    car(a3);
    cdr(a3);
}/>

<eval DEBUG{
       LEN(235);
       LEN("abc");
       LEN(1 2 3);
       LEN(2,3,4,5);
}/>


<eval DEBUG{
   amap=MAP(_,_);
   bmap=MAP{"abc"=>3,"def"=>4};
   cmap=MAP{};
   amap["abc"] = 3;
   amap[4] = "def";
   abc = amap["abc"];

   LEN(amap) ;
   foreach (amap :  (CLEAR from, CLEAR to) : FALSE) {
       (from) "=>" (to);
   }
} />

<eval  DEBUG{
    input="bc ade lkd";
    input : STRING;
    345 : STRING;
    input : INT;
    345 : INT;
    345 : (INT | STRING);
    345 : (INT _);
    input : MAP(_,_);
    amap : MAP(_,_);
    amap : MAP(INT,_);
    input : (_ _);
    (2 3) : (_ _);
    (2,3) : (_,_);
    (3 2) : (first second);
    SPLIT("",input) => var=(STRING STRING STRING); 
    SPLIT("",input) => var=(STRING,STRING,STRING); 
}/>

<eval  DEBUG{
    "abc" ^ "def" ^ 3;
    ("abc" "def") ^ 3 ;
    input="bc,ade,lkd";
    SPLIT (1, "abc");
    SPLIT(",",input);
    SPLIT(",", MyCodeTemplate#input);
}/>

<eval DEBUG{
    myFunc = DELAY { x = 100; } ;
}/>

