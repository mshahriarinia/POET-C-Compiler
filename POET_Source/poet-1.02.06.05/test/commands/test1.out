
DEBUGGING at level 1
"3":STRING
===>1
3:STRING
===>0
"3":ID
===>0
"A3":ID
===>1
MyCodeTemplate#"123":STRING
===>0
MyCodeTemplate#123:MyCodeTemplate
===>1
3:MyCodeTemplate
===>0
MyCodeTemplate#123:MyCodeTemplate#INT
===>1
("abc" "." "ext"):STRING
===>0
("3" "4" "5"):(INT)...
===>1
3:(INT)...
===>1
(3 4 5):(INT)...
===>1
(3 4 5 "abc"):(INT)...
===>0
(3 4 5 "abc"):(_)...
===>1
3:0..2
===>0
3:0..5
===>1
"a":0..5
===>0
"a":CODE
===>0
MyCodeTemplate:CODE
===>1
MyCodeTemplate#123:CODE
===>1
("abc" "." "ext"):CODE
===>0
MyCodeTemplate#123:XFORM
===>0
("abc" "." "ext"):XFORM
===>0
foo:XFORM
===>1
"abc":TUPLE
===>0
("abc",2):TUPLE
===>1
MyCodeTemplate#123:TUPLE
===>0
END DEBUGGING at level 1

DEBUGGING at level 1
"3"=>STRING
===>"3"
a=("3" "5")
===>("3" "5")
a=>STRING
===>"35"
3=>STRING
===>"3"
("abc" "." "ext")=>STRING
===>"abc.ext"
"3"=>INT
===>3
"3"=>INT=>STRING
===>"3"
END DEBUGGING at level 1

DEBUGGING at level 1
a1=(3 4 5)
===>(3 4 5)
a2=(1 2 a1)
===>(1 2 (3 4 5))
a3=(1 2)::a1
===>((1 2) 3 4 5)
a4=1::2::a1
===>(1 2 3 4 5)
HEAD(a2)
===>1
TAIL(a2)
===>(2 (3 4 5))
HEAD(a3)
===>(1 2)
TAIL(a3)
===>(3 4 5)
END DEBUGGING at level 1

DEBUGGING at level 1
LEN(235)
===>1
LEN("abc")
===>3
LEN((1 2 3))
===>3
LEN((2,3,4,5))
===>4
END DEBUGGING at level 1

DEBUGGING at level 1
amap=MAP(_,_)
===>MAP{}
bmap=MAP((("abc" 3) ("def" 4)))
===>MAP{"abc"=>(3);"def"=>(4);}
cmap=MAP("")
===>MAP{}
amap["abc"]=3
===>3
amap[4]="def"
===>"def"
abc=amap["abc"]
===>3
LEN(amap)
===>2
Entering FOREACH(MAP{"abc"=>3;4=>"def";})
FOREACH found : ("abc",3)
FOREACH found : (4,"def")
Exiting FOREACH(MAP{"abc"=>3;4=>"def";})
END DEBUGGING at level 1

DEBUGGING at level 1
input="bc ade lkd"
===>"bc ade lkd"
input:STRING
===>1
345:STRING
===>0
input:INT
===>0
345:INT
===>1
345:INT|STRING
===>1
345:(INT _)
===>0
input:MAP(_,_)
===>0
amap:MAP(_,_)
===>1
amap:MAP(INT,_)
===>0
input:(_ _)
===>0
(2 3):(_ _)
===>1
(2,3):(_,_)
===>1
(3 2):(first second)
===>1
SPLIT("",input)
===>("bc" " " "ade" " " "lkd")
SPLIT("",input)=>var=(STRING STRING STRING)
===>("bc" "ade" "lkd")
SPLIT("",input)
===>("bc" " " "ade" " " "lkd")
SPLIT("",input)=>var=(STRING,STRING,STRING)
===>("bc","ade","lkd")
END DEBUGGING at level 1

DEBUGGING at level 1
"abc"^"def"
===>"abcdef"
"abc"^"def"^3
===>"abcdef3"
("abc" "def")^3
===>"abcdef3"
input="bc,ade,lkd"
===>"bc,ade,lkd"
SPLIT(1,"abc")
===>("a" "bc")
SPLIT(",",input)
===>("bc" "," "ade" "," "lkd")
SPLIT(",",MyCodeTemplate#input)
===>MyCodeTemplate#(("bc" "," "ade" "," "lkd"))
END DEBUGGING at level 1

DEBUGGING at level 1
myFunc=DELAY{x=100}
===>x=100
END DEBUGGING at level 1
DEBUGGING result: myFunc=DELAY{x=100}===>POETEvalExp:x=100
